File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/decorator/lab/CoffeeWithMilkWithSugar.m
% CoffeeWithMilkWithSugar.m

classdef CoffeeWithMilkWithSugar < CoffeeWithMilk
	methods
		function prepare(theCoffeeWithMilkWithSugar)
			prepare@CoffeeWithMilk(theCoffeeWithMilkWithSugar);
			fprintf('Adding sugar.\n');
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/decorator/lab/coffee_with_milk_with_sugar.cpp
// coffee_with_milk_with_sugar.cpp

#include <iostream>
#include "coffee_with_milk.h"
#include "coffee_with_milk_with_sugar.h"

void coffee_with_milk_with_sugar::prepare()
{
	coffee_with_milk::prepare();
	std::cout << "Adding sugar.\n";
}
